docs/source/user_guide/user_guide/sub_functions.md

```diff
@@ -0,0 +1,11 @@
+# Sub-functions
+
+A @ti.kernel can call other functions, as long as those functions have an appropriate taichi annotation.
+
+## ti.func
+
+@ti.func is the standard annotation for a function that can be called from a kernel. They can also be called from other @ti.func's. @ti.func is inlined into the calling kernel.
+
+## ti.real_func
```

Comment:
What about `@ti.pure` ?

Reply:
Not merged yet. But perahps I should create doc in the pr

Reply:
like, in the caching pr.